src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -719,60 +976,60 @@ CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & clust
 {
     using namespace Clusters::NetworkCommissioning::Commands;
 
-    constexpr CommandId acceptedCommandsListWiFi[] = {
-        ScanNetworks::Id, AddOrUpdateWiFiNetwork::Id, RemoveNetwork::Id, ConnectNetwork::Id, ReorderNetwork::Id,
-    };
-    constexpr CommandId acceptedCommandsListThread[] = {
-        ScanNetworks::Id, AddOrUpdateThreadNetwork::Id, RemoveNetwork::Id, ConnectNetwork::Id, ReorderNetwork::Id,
-    };
-
     if (mFeatureFlags.Has(Feature::kThreadNetworkInterface))
     {
-        for (const auto & cmd : acceptedCommandsListThread)
+        for (auto && cmd : {
+                 ScanNetworks::Id,
+                 AddOrUpdateThreadNetwork::Id,
+                 RemoveNetwork::Id,
+                 ConnectNetwork::Id,
+                 ReorderNetwork::Id,
+             })
         {
-            if (callback(cmd, context) != Loop::Continue)
-            {
-                break;
-            }
+            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
         }
-
-        return CHIP_NO_ERROR;
     }
-
-    if (mFeatureFlags.Has(Feature::kWiFiNetworkInterface))
+    else if (mFeatureFlags.Has(Feature::kWiFiNetworkInterface))
     {
-        for (const auto & cmd : acceptedCommandsListWiFi)
+        for (auto && cmd : {
+                 ScanNetworks::Id,
+                 AddOrUpdateWiFiNetwork::Id,
+                 RemoveNetwork::Id,
+                 ConnectNetwork::Id,
+                 ReorderNetwork::Id,
+             })
         {
-            if (callback(cmd, context) != Loop::Continue)
-            {
-                break;
-            }
+            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
         }
+    }
 
-        return CHIP_NO_ERROR;
+    if (mFeatureFlags.Has(Feature::kPerDeviceCredentials))
```

Comment:
But this should only happen if kWiFiNetworkInterface is enabled, no?  Why is this not inside that block?

Reply:
Doesn't really make a difference either way... the `kPerDeviceCredentials` is only ever set if `WiFi` is also set. In other places (e.g. handling commands) I'm only checking for the one flag, so it seems more consistent to do that everywhere than to always check for both.